bundles/io/org.eclipse.smarthome.io.transport.serial/src/main/java/org/eclipse/smarthome/io/transport/serial/internal/SerialPortManagerImpl.java

```diff
@@ -50,6 +50,7 @@ protected void unsetSerialportRegistry(SerialPortRegistry registry) {
     @Override
     public Stream<SerialPortIdentifier> getIdentifiers() {
         if (registry == null) {
+            logger.warn("SerialPortRegistry is not set; no SerialPortIdentifier found");
```

Comment:
Imho this can be considered an illegal state, i.e. rather a bug situation. In that case, I would suggest to add the exception as a parameter, so that the stacktrace is at hand when that happens.

Reply:
But there is no exception in this case so I cannot log it.